### PR TITLE
Adding ClientConnectionError handler

### LINKED ISSUE #128: Don't produce errors when device is unreachable
## Describe the bug

When device is turned off the component shouldn't spam with errors like this, we can add a message on `debug` level about this. But authentication errors (when token is invalid) are still worth an error message.

## Logs

`2021-03-31 10:59:33 ERROR (MainThread) [custom_components.google_home] Request error: Cannot connect to host 192.168.XXX.XX:8443 ssl:default [None]`

```
Logger: custom_components.google_home
Source: custom_components/google_home/api.py:162
Integration: Google Home (documentation, issues)
First occurred: 30 maart 2021 9:03:53 (4754 occurrences)
Last logged: 11:18:53

Request error: Cannot connect to host 192.168.XXX.XX:8443 ssl:default [Connect call failed ('192.168.XXX.X', 8443)]
Request error: Cannot connect to host 192.168.XXX.XX:8443 ssl:default [Connect call failed ('192.168.XXX.XX', 8443)]
Request error: [Errno 104] Connection reset by peer
Request error: Cannot connect to host 192.168.XXX.XX:8443 ssl:default [None]
Request error: Cannot connect to host 192.168.XXX.XX:8443 ssl:default [Connect call failed ('192.168.XXX.XX', 8443)]
```

### PR DESCRIPTION
**Pull request by leikoilja**
Closes #128,

We should not flood HA logs with errors if we can't connect to the device (happens when some devices go offline). Therefore let's catch the ConnectionError and log as `debug`.

I have tested locally by fetching data from device and then unplugging it

### CODE REVIEW

File: custom_components/google_home/api.py [hunk 1]
```diff
@@ -142,12 +143,12 @@ async def get_alarms_and_timers(
                     # We need to retry the update task instead of just cleaning the list
                     self.google_devices = []
                 elif response.status == HTTP_NOT_FOUND:
-                    device.available = False
```
Comment (leikoilja) [inline custom_components/google_home/api.py:145]: flip, that's from a different PR. Will remove rn

**Review by KapJI — APPROVED** [review]
Looks good, thanks! 🎉

File: custom_components/google_home/api.py [hunk 2]
```diff
@@ -146,8 +147,9 @@ async def get_alarms_and_timers(
                     _LOGGER.debug(
                         (
                             "Failed to fetch data from %s, API returned %d. "
-                            "The device(hardware='%s') is not Google Home "
-                            "compatable and has no alarms/timers."
+                            "The device(hardware='%s') is possibly not Google Home "
+                            "compatable and has no alarms/timers. "
+                            "Will retry later."
```
Comment (KapJI) [inline custom_components/google_home/api.py:152]: This also looks unrelated to this PR.

Comment (leikoilja) [reply custom_components/google_home/api.py:152]: Ops, you re right, thanks
